openlibrary/templates/type/edition/view.html

```diff
@@ -500,18 +500,32 @@ <h4>$:_('Links <span class="gray small sansserif">outside Open Library</span>')<
         $# pages like /books/ia:foo00bar are fake records created from metadata API.
         $# Adding them to lists doesn't work. Disabling it to avoid any issues.
         $if "lists" in ctx.features and not page.is_fake_record():
-          <h2 class="list-heading"><a href="$page.url()/lists">$_('Lists containing this Book')</a></h2>
+          <div class="lists-heading">
+            <h2 class="lists-title">$_('Lists')</h2>
+            $if ctx.user:
+              <a href="$page.url()/lists">$_('See All')</a>
+            $else:
+              <a href="$homepath()/lists">$_('Learn More')</a>
```

Comment:
```suggestion
            $ has_lists = False
            $if (work and work.get_lists()) or (edition and edition.get_lists()):
              $ has_lists = True
            $if has_lists:
              <a href="$page.url()/lists">$_('See All')</a>
```
I'm not sure that there's a need to redirect to the main list page if the patron is not logged in.  It may also be a bit confusing to have a "See all" link when it says that the work is not on any list.
Let's set the `has_lists` flag here, and only display the "See All" link if the book appears on any list.  Note that Python lists are considered falsey if they are empty, and truthy otherwise.